hooks/tk-adobecc_actions.py

```diff
@@ -0,0 +1,222 @@
+                                     "description": "This will open the file."})
+
+        return action_instances
+
+    def execute_multiple_actions(self, actions):
+        """
+        Executes the specified action on a list of items.
+
+        The default implementation dispatches each item from ``actions`` to
+        the ``execute_action`` method.
+
+        The ``actions`` is a list of dictionaries holding all the actions to
+        execute.
+
+        Each entry will have the following values:
+
+            name: Name of the action to execute
+            sg_publish_data: Publish information coming from Shotgun
+            params: Parameters passed down from the generate_actions hook.
+
+        .. note::
+            This is the default entry point for the hook. It reuses the
+            ``execute_action`` method for backward compatibility with hooks
+            written for the previous version of the loader.
+
+        .. note::
+            The hook will stop applying the actions on the selection if an error
+            is raised midway through.
+
+        :param list actions: Action dictionaries.
+        """
+        for single_action in actions:
+            name = single_action["name"]
+            sg_publish_data = single_action["sg_publish_data"]
+            params = single_action["params"]
+            self.execute_action(name, params, sg_publish_data)
+
+    def execute_action(self, name, params, sg_publish_data):
+        """
+        Execute a given action. The data sent to this be method will
+        represent one of the actions enumerated by the generate_actions method.
+        
+        :param name: Action name string representing one of the items returned
+                     by generate_actions.
+        :param params: Params data, as specified by generate_actions.
+        :param sg_publish_data: Shotgun data dictionary with all the standard
+                                publish fields.
+        """
+        app = self.parent
+        app.log_debug("Execute action called for action %s. "
+                      "Parameters: %s. Publish Data: %s" % (name, params, sg_publish_data))
+
+        # resolve path
+        path = self.get_publish_path(sg_publish_data)
+
+        if not os.path.exists(path):
+            raise Exception("File not found on disk - '%s'" % path)
+
+        if name == _OPEN_FILE:
+            self._open_file(path, sg_publish_data)
+        if name == _ADD_AS_A_LAYER:
+            self._place_file(path, sg_publish_data)
+
+    ###########################################################################
+    # helper methods
+
+    def _open_file(self, path, sg_publish_data):
+        """
+        Import contents of the given file into the scene.
+        
+        :param path: Path to file.
+        :param sg_publish_data: Shotgun data dictionary with all the standard
+                                publish fields.
+        """
+        path = "/".join(path.split(os.path.sep))
```

Comment:
More likely is the inability/difficulty in properly escaping a backslash due to the number of times the string is going to be interpreted as it travels across various layers of the RPC interface.